test/test.js

```diff
@@ -0,0 +1,276 @@
+const chai = require('chai');
+const chaiHttp = require('chai-http');
+
+chai.use(chaiHttp);
+const { expect } = chai;
+
+const karenEndpoint = process.env.HOST;
+const createdUsers = new Set();
+
+const preExistingUser = {
+  name: 'Pre-existing User',
+  email: 'preexistinguser@foo.bar',
+  password: 'foobar',
+};
+
+// Create a pre-existing user that each test can use as it needs.
+beforeEach(async () => {
+  const res = await chai.request(karenEndpoint)
+    .post('/karen/v1/users')
+    .send(preExistingUser);
+
+  expect(res).to.have.status(201);
+  preExistingUser.id = res.body.id;
+  createdUsers.add(res.body.id);
+});
+
+// Delete any leftover created users to allow for a fresh starting point for the
+// next test.
+afterEach(async () => {
+  const requests = [];
+  Array.from(createdUsers).forEach((userID) => {
+    requests.push(chai.request(karenEndpoint)
+      .delete('/karen/v1/users/self')
+      .set('User-ID', userID));
+    createdUsers.delete(userID);
+  });
+
+  const responses = await Promise.all(requests);
+  responses.forEach((res) => {
+    expect(res).to.have.status(204);
+  });
+});
+
+describe('POST /karen/v1/users', () => {
+  it('should create a user', async () => {
+    const res = await chai.request(karenEndpoint)
+      .post('/karen/v1/users')
+      .send({
+        name: 'Foo Bar',
+        email: 'foo@bar.baz',
+        password: 'foobarbaz',
+      });
+
+    expect(res).to.have.status(201);
+    expect(res.body).to.have.keys('id', 'name', 'email');
```

Comment:
The chai docs recommends using `all.keys` even though just `keys` defaults to `all.keys` just to help with readability.
```suggestion
    expect(res.body).to.have.all.keys('id', 'name', 'email');
```

Reply:
Yeah, forgot about that.

Reply:
I just know because I had to go to the docs to check how the `keys` thing works haha.